packages/Search/src/details/DTK_DetailsTmp.hpp

```diff
@@ -0,0 +1,690 @@
+    send_across_network( std::shared_ptr<Tpetra::Distributor> distributor,
+                         Kokkos::View<T *, DeviceType> exports,
+                         Kokkos::View<T *, DeviceType> imports );
+
+    static double epsilon;
+};
+
+template <typename DeviceType>
+double Tmp<DeviceType>::epsilon = 1.0e-6;
+
+template <typename DeviceType>
+template <typename T>
+void Tmp<DeviceType>::send_across_network(
+    std::shared_ptr<Tpetra::Distributor> distributor,
+    Kokkos::View<T *, DeviceType> exports,
+    Kokkos::View<T *, DeviceType> imports )
+{
+    auto exports_host = Kokkos::create_mirror_view( exports );
+    Kokkos::deep_copy( exports_host, exports );
+    auto imports_host = Kokkos::create_mirror_view( imports );
+    distributor->doPostsAndWaits(
+        Teuchos::ArrayView<T const>( exports_host.data(),
+                                     exports_host.extent( 0 ) ),
+        1, Teuchos::ArrayView<T>( imports_host.data(),
+                                  imports_host.extent( 0 ) ) );
+    Kokkos::deep_copy( imports, imports_host );
+}
+
+template <typename DeviceType>
+template <typename Query>
+void Tmp<DeviceType>::make_plan( Point epsilon,
+                                 Kokkos::View<Query *, DeviceType> queries,
+                                 BVH<DeviceType> const &bvh,
+                                 Kokkos::View<int *, DeviceType> &indices,
+                                 Kokkos::View<int *, DeviceType> &offset )
+{
+    int const n_queries = queries.extent( 0 );
+    Kokkos::View<Details::Overlap *, DeviceType> overlap_queries( "noname",
+                                                                  n_queries );
+
+    Kokkos::parallel_for( REGION_NAME( "noname" ),
+                          Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
+                          KOKKOS_LAMBDA( int q ) {
+                              Point point = queries( q )._query_point;
+                              Box box( {
+                                  point[0] - epsilon[0], point[0] + epsilon[0],
+                                  point[1] - epsilon[1], point[1] + epsilon[1],
+                                  point[2] - epsilon[2], point[2] + epsilon[2],
+                              } );
+                              overlap_queries( q ) = Details::Overlap( box );
+                          } );
+    Kokkos::fence();
+
+    bvh.query( overlap_queries, indices, offset );
+}
+
+template <typename DeviceType>
+template <typename Query>
+void Tmp<DeviceType>::query_dispatch(
+    DistributedSearchTree<DeviceType> const &tree,
+    Kokkos::View<Query *, DeviceType> queries,
+    Kokkos::View<int *, DeviceType> &indices,
+    Kokkos::View<int *, DeviceType> &offset,
+    Kokkos::View<int *, DeviceType> &ranks, Details::NearestPredicateTag )
+{
+    //\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\//
+    //\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\//
```

Comment:
I was using backslash to highlights pieces of code that were different in the nearest query version of the search.